devimint/src/federation.rs

```diff
@@ -241,6 +249,22 @@ impl Federation {
         Ok(())
     }
 
+    pub async fn degrade_federation(&mut self, process_mgr: &ProcessManager) -> Result<()> {
```

Comment:
Why is this fn pub? ~Seems like an internal one that's only called on initialization?~ I see it's also used for one test, so maybe also useful for external devimint users? @maan2003 @justinmoon

Reply:
Good question, I think we can make private or remove altogether. It was useful in the first commit where backwards-compatible tests were run in a degraded state and we had to handle the 0.2.1 bug by starting all peers for certain code paths then degrading after.

https://github.com/fedimint/fedimint/commit/7e288aae56678faecf085384906ddacc327bd79c#diff-75cef4315751be4fa6703860684a38e9c382d522c05f6df275846e94a373562dR555

Reply:
In general I think almost everything in devimint should be `pub`. External module developers will want the full power of devimint when constructing tests.

Reply:
I kept `pub` along with the sanity check we have enough nodes to reach consensus